hypershift-operator/controllers/hostedcluster/metrics_test.go

```diff
@@ -153,6 +211,16 @@ func TestMetrics(t *testing.T) {
 				hostedClusterInitialRolloutDuration.WithLabelValues(cluster.Namespace, cluster.Name).Set(*versionRolloutTime)
 			}
 
+			for _, condition := range cluster.Status.Conditions {
+				if condition.Type == string(hyperv1.ValidAWSIdentityProvider) && condition.Status == metav1.ConditionFalse {
+					platformaws.SkippedCloudResourcesDeletion.WithLabelValues(cluster.Namespace, cluster.Name).Set(float64(1))
```

Comment:
this doesn't make sense from a unit test pov. You're setting the values your self, so you'll always match your expectation.
You need to call here the functions that set the metric under different circumstance dictated by the test cases, just like we do for the other metrics in this unit test, i.e `DeleteOrphanedMachines` and `deleteAWSEndpointServices`

Also if that's difficult please consider and e2e test as suggested in the original comment.

Reply:
Maybe this could fit better in a E2E test but I will try to make it in unit ones.